### PR TITLE
Added keptnContext query parameter to sequence state GET endpoint

### LINKED ISSUE #4433: Additional parameter keptnContext for sequence endpoint
Because of the deep-links in the bridge, we need an additional parameter `keptnContext` for the sequence endpoint in order to load the sequence that is specified in the deep link.

## Acceptance Criteria
- [ ] Sequence State endpoint provides additional `keptnContext` parameter to fetch sequence states

## Definition of done
- [ ] swagger doc up to date
- [ ] Integration tests are updated

### PR DESCRIPTION
**Pull request by warber**
closes #4433 

I've introduced an additional query parameter go fetch sequence enpoints by keptn context.
Note, that I've named the paramter `keptnContext` instead of `shkeptncontext`.

I would also propose to switch to just `keptnContext` in other places as well.
This needs to be backwards compatible, though.

Please comment your opinion on that.

Signed-off-by: warber <bernd.warmuth@dynatrace.com>

### CODE REVIEW

File: shipyard-controller/docs/docs.go [hunk 1]
```diff
@@ -2033,7 +1999,7 @@ type swaggerInfo struct {
 
 // SwaggerInfo holds exported Swagger Info so clients can modify it
 var SwaggerInfo = swaggerInfo{
-	Version:     "1.0",
+	Version:     "develop",
```
Comment (thisthat) [inline shipyard-controller/docs/docs.go:2002]: is this correct?

File: shipyard-controller/docs/swagger.yaml [hunk 2]
```diff
@@ -533,7 +514,7 @@ info:
     name: Apache 2.0
     url: http://www.apache.org/licenses/LICENSE-2.0.html
   title: Control Plane API
-  version: "1.0"
+  version: develop
```
Comment (thisthat) [inline shipyard-controller/docs/swagger.yaml:517]: same question here

File: shipyard-controller/handler/sequence_migrator.go [hunk 3]
```diff
@@ -82,9 +82,9 @@ func (sm *SequenceMigrator) migrateSequence(projectName string, rootEvent models
 	log.Infof("checking if root event for shkeptncontext %s already has a task sequence state in the collection", rootEvent.Shkeptncontext)
 	sequence, err := sm.taskSequenceRepo.FindSequenceStates(models.StateFilter{
 		GetSequenceStateParams: models.GetSequenceStateParams{
-			Project: projectName,
+			Project:        projectName,
+			KeptnContext: rootEvent.Shkeptncontext,
```
Comment (thisthat) [inline shipyard-controller/handler/sequence_migrator.go:86]:
```suggestion
			Project:      projectName,
			KeptnContext: rootEvent.Shkeptncontext,
```

File: shipyard-controller/handler/sequencehooks/sequencestate.go [hunk 4]
```diff
@@ -123,9 +123,9 @@ func (smv *SequenceStateMaterializedView) findSequenceStateForEvent(event models
 
 	return smv.SequenceStateRepo.FindSequenceStates(models.StateFilter{
 		GetSequenceStateParams: models.GetSequenceStateParams{
-			Project: eventScope.Project,
+			Project:        eventScope.Project,
+			KeptnContext: eventScope.KeptnContext,
```
Comment (thisthat) [inline shipyard-controller/handler/sequencehooks/sequencestate.go:127]:
```suggestion
			Project:      eventScope.Project,
			KeptnContext: eventScope.KeptnContext,
```

File: shipyard-controller/handler/sequencehooks/sequencestate.go [hunk 5]
```diff
@@ -199,9 +199,9 @@ func (smv *SequenceStateMaterializedView) updateLastEventOfSequence(event models
 
 	states, err := smv.SequenceStateRepo.FindSequenceStates(models.StateFilter{
 		GetSequenceStateParams: models.GetSequenceStateParams{
-			Project: eventScope.Project,
+			Project:        eventScope.Project,
+			KeptnContext: eventScope.KeptnContext,
```
Comment (thisthat) [inline shipyard-controller/handler/sequencehooks/sequencestate.go:203]:
```suggestion
			Project:      eventScope.Project,
			KeptnContext: eventScope.KeptnContext,
```

File: shipyard-controller/docs/swagger.yaml (hunk 2 above)
Comment (warber) [reply shipyard-controller/docs/swagger.yaml:517]: yes, it gets replaced here https://github.com/keptn/keptn/blob/12c9f275ad71c16923b9cc7eed700ec05c1ecb49/shipyard-controller/Dockerfile#L40

**Review by thisthat — APPROVED** [review]

**Review by bacherfl — APPROVED** [review]

**Pull request merged**